### ISSUE TITLE
future feature annotations syntax error when import package

### ISSUE THREAD

**Issue by Georift** [fastai/fastai#3613]
Please confirm you have the latest versions of fastai, fastcore, and nbdev prior to reporting a bug (delete one): **YES**

**Describe the bug**

When running fastai version [v2.5.5](https://github.com/fastai/fastai/releases/tag/2.5.5) on Python 3.6, I'm seeing a SyntaxError thrown while importing packages.

Trying the same reproduction again but with fastai v2.5.4 the error is no longer present.

Could it be something related to #3601?

**To Reproduce**
```bash
docker run --rm -v ~/.config/pip:/root/.cache/pip python:3.6.13-slim /bin/sh -c "pip install fastai==2.5.5 && python -c 'import fastai; from fastai.vision import augment'"
```

**Expected behavior**
The command above to run without error

**Error with full stack trace**
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/fastai/vision/augment.py", line 12, in <module>
    from ..data.all import *
  File "/usr/local/lib/python3.6/site-packages/fastai/data/all.py", line 1, in <module>
    from ..torch_basics import *
  File "/usr/local/lib/python3.6/site-packages/fastai/torch_basics.py", line 9, in <module>
    from .imports import *
  File "/usr/local/lib/python3.6/site-packages/fastai/imports.py", line 30, in <module>
    from fastcore.all import *
  File "/usr/local/lib/python3.6/site-packages/fastcore/all.py", line 3, in <module>
    from .dispatch import *
  File "/usr/local/lib/python3.6/site-packages/fastcore/dispatch.py", line 4
    from __future__ import annotations
    ^
SyntaxError: future feature annotations is not defined

```

**Additional context**

The following command runs without issue:
```
docker run --rm -v ~/.config/pip:/root/.cache/pip python:3.6.13-slim /bin/sh -c "pip install fastai==2.5.4 && python -c 'import fastai; from fastai.vision import augment'"
```

**Comment by malisit** [comment]
I have the same issue. I can't use fastai on Python 3.6 because of the annotations import which was available with Python 3.7 and onwards.

**Comment by jph00** [comment]
Python 3.6 is EOL and not supported in fastai.
